lib/assessment.rb

```diff
@@ -43,6 +59,13 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    #I don't really understand the ARGV[0], just took from read-file.rb demo yesterday, and I don't know when to use it/if I need it here.
+    File.open(ARGV[0]) do |filename|
+      filename.reduce(&:+)
+      filename.gsub(/\n/, ' ')
+    end
```

Comment:
Please disregard my previous notes on this question - I was mistaken about something with `File.open`. As written, it's still not quite right (replacing newlines with spaces should happen before concatenation, and you still need to strip off whitespace at the end); however, I'm giving you half-credit on this one now.